docs/external_metrics.md

```diff
@@ -0,0 +1,420 @@
+# Rezolus External Metrics Specification
+
+## Version 1.0
+
+## Overview
+
+This specification defines a custom binary file format for ingesting external
+metrics into Rezolus via memory-mapped files. The format is designed for
+high-performance, lock-free communication between a producer process (writing
+metrics) and a consumer process (Rezolus agent) using `mmap()` on Linux systems.
+
+## Design Goals
+
+- **Zero-copy access**: Direct memory mapping for maximum performance
+- **Lock-free reads**: Consumer can read without blocking producer
+- **Simple serialization**: No external dependencies required
+- **Concurrent safety**: Safe for one writer, multiple readers
+- **Static metric set**: Metrics defined at file creation, no runtime additions
+
+## File Structure
+
+The file consists of three main sections:
+
+```
+┌─────────────────┐
+│     Header      │
+├─────────────────┤
+│ Metrics Catalog │
+├─────────────────┤
+│   Data Section  │
+└─────────────────┘
+```
+
+## Header Section
+
+The header is exactly 64 bytes and contains:
+
+| Offset | Size | Field | Description |
+|--------|------|-------|-------------|
+| 0 | 4 | Magic | Magic number: `0x52455A4C` ("REZL") |
+| 4 | 1 | Version Major | Major version number (current: 1) |
+| 5 | 1 | Version Minor | Minor version number (current: 0) |
+| 6 | 1 | Status | Control byte (bits 0-1: ready flags) |
+| 7 | 1 | Reserved | Must be zero |
+| 8 | 4 | Metric Count | Number of metrics in catalog (u32, max 1024) |
+| 12 | 4 | Catalog Size | Size of catalog section in bytes (u32) |
+| 16 | 1 | Checksum Type | Checksum algorithm (0=None, 1=CRC32-IEEE) |
+| 17 | 3 | Reserved | Must be zero |
+| 20 | 4 | Catalog Checksum | CRC32 of catalog section (0 if type=0) |
+| 24 | 8 | Data Offset | Offset to data section (u64) |
+| 32 | 8 | Data Size | Size of data section in bytes (u64) |
+| 40 | 8 | Timestamp | Unix timestamp of file creation (u64) |
+| 48 | 16 | Reserved | Must be zero |
+
+### Status Byte Details
+
+- **Bit 0 (Catalog Ready)**: Set to 1 when catalog is complete and checksum
```

Comment:
Simplify ready flags as a single bit - specify that data section should be initialized, catalog must be populated and sync must have happened before setting ready flag